Functions/ImageProcessing/xASL_im_MergePWI4D.m

```diff
@@ -79,39 +79,48 @@
+		% Some fields are concatenated across sessions. Other fields are given per entire sequence and not per volume. So the fields are saved only once for the first session
+		% and with following session, we check for consistence and report errors
+
+		% A list of fields
+		listFields2Merge = {'MagneticFieldStrength', 'PulseSequenceType', 'MRAcquisitionType', 'Manufacturer', 'ArterialSpinLabelingType', ...% These are checked but not concatenated
+			                'LabelingDuration', 'PostLabelingDelay', 'EchoTime'};% These are concatenated
+
+		% An indication of a field is merged or concatenated
+		bFields2Concatenate = [0, 0, 0, 0, 0, ...
+		                       1, 1, 1];
+
+		% Go through all the fields to merge
+		for iField2Merge = 1:length(listFields2Merge)
+			% The field exists and is not empty, we therefore need to check if we can merge it across sessions
+ 			if isfield(jsonPWI4Dcurrent, listFields2Merge{iField2Merge}) && ~isempty(jsonPWI4Dcurrent.(listFields2Merge{iField2Merge}))
+				if iSession == 1 
+					% For the first session, just add the field
+					jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}) = jsonPWI4Dcurrent.(listFields2Merge{iField2Merge});
+				elseif ~isfield(jsonPWI4DConcatenated, listFields2Merge{iField2Merge})
+					% Session > 1, but the field was not added for session 1, we report a conflict
+					error(['Cannot merge sessions as the JSON field ' listFields2Merge{iField2Merge} ' is missing for session ' x.modules.asl.sessionsToMerge{1} ' but not for session ' x.modules.asl.sessionsToMerge{iSession}]);
+				else
+					% The field seems to be consistently present/missing/empty on both sessions. The field is either check for consistency across sessions or concatenated
+					if bFields2Concatenate(iField2Merge)
+						% Concatenation
+						jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}) = [jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}); jsonPWI4Dcurrent.(listFields2Merge{iField2Merge})];
+					else
+						% Just checking consistency
+						if ~isequal(jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}), jsonPWI4Dcurrent.(listFields2Merge{iField2Merge}))
+							% Otherwise (field added for first session and we are now adding more) check for consistency and report an error if non-consistent
+							error(['Cannot merge sessions as the JSON field ' listFields2Merge{iField2Merge} ' differs between sessions ' x.modules.asl.sessionsToMerge{iSession} ' and ' x.modules.asl.sessionsToMerge{1}]);
+						end
+					end
+				end
+			elseif isfield(jsonPWI4DConcatenated, listFields2Merge{iField2Merge}) && ~isempty(jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}))
+				% Field is missing for the current session, but it was added for the first session
+				error(['Cannot merge sessions as the JSON field ' listFields2Merge{iField2Merge} ' is missing for session ' x.modules.asl.sessionsToMerge{iSession} ' but not for session ' x.modules.asl.sessionsToMerge{1}]);
```

Comment:
Again, we don't care for which session the field is NOT missing, but only which session the field IS missing. And the path! (Or subject ID)

Reply:
SubjectID added.

We mention both missing and non-missing sessions. As said above -> we are checking consistency, so we need to report both.